src/App.test.js

```diff
@@ -1,8 +1,86 @@
-import { render, screen } from '@testing-library/react';
-import App from './App';
+import { render, screen } from "@testing-library/react";
+import userEvent from "@testing-library/user-event";
+import { createStore } from "redux";
 
-test('renders learn react link', () => {
-  render(<App />);
-  const linkElement = screen.getByText(/learn react/i);
-  expect(linkElement).toBeInTheDocument();
+import App from "./App";
+import { reducer } from "./store/index.js";
+
+const renderApp = () => {
+  const store = createStore(reducer);
+  render(<App store={store} />);
```

Comment:
If you create an actual redux store and pass it to a component manually, this test is probably redundant, and almost certainly not needed - because it's redundant with your presentation component tests.

Rule of thumb for testing React components:

Given certain props, make sure correct data gets rendered.  Ignore the store. Just look at prop input and component output.